### PR TITLE
Merge from xircuits repo

### PR DESCRIPTION
**Pull request by liadomide**
- [x] python 3.11, Jupyter 4
- [x] merge release 1.13.1 from upstream

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/feature-request.md [hunk 1]
```diff
@@ -7,6 +7,9 @@ assignees: ''
 
 ---
 
+**Xircuits Version**
```
Comment (liadomide) [inline .github/ISSUE_TEMPLATE/feature-request.md]: The entire folder `ISSUE_TEMPLATE` does not seem to be applying to our use-case.
Maybe remove it from our branch ?

File: .github/workflows/binder-badge.yaml [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  pull_request_target:
+    types: [opened]
+
+jobs:
+  binder:
+    runs-on: ubuntu-latest
+    permissions:
+      pull-requests: write
+    steps:
+    - name: comment on PR with Binder link
```
Comment (liadomide) [inline .github/workflows/binder-badge.yaml]: This workflow does not seem to apply to our build. Maybe remove it

File: .github/workflows/pypi-github-release-publish.yml [hunk 3]
```diff
@@ -8,9 +8,9 @@ jobs:
   deploy:
     runs-on: ubuntu-latest
     steps:
-    - uses: actions/checkout@v2
+    - uses: actions/checkout@v4
     - name: Set up Python
-      uses: actions/setup-python@v2
+      uses: actions/setup-python@v5
       with:
         python-version: '3.8'
```
Comment (liadomide) [inline .github/workflows/pypi-github-release-publish.yml]: switch to `python 3.11`

File: .github/workflows/run-python-tests.yml [hunk 4]
```diff
@@ -15,42 +16,50 @@ jobs:
```
Comment (liadomide) [inline .github/workflows/run-python-tests.yml:16]: I suggest ` python-version: [ "3.8", "3.11", "3.12" ]` (3.8 to be soon removed)

File: .github/workflows/run-ui-tests.yml [hunk 5]
```diff
@@ -1,73 +1,87 @@
...
 
 jobs:
-  build:
+  build_and_test:
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        python-version: ["3.8","3.9","3.10", "3.11"]
+        python-version: ["3.8", "3.9", "3.10", "3.11"]
```
Comment (liadomide) [inline .github/workflows/run-ui-tests.yml]: Same here `3.8, 3.11 and 3.12`

File: binder/environment.yml [hunk 6]
```diff
@@ -0,0 +1,20 @@
+name: xircuits-demo
+
+channels:
+  - conda-forge
+
+dependencies:
+  # Runtime dependencies installed through conda
+  - python >=3.8,<3.11
```
Comment (liadomide) [inline binder/environment.yml]: could we try `<=3.12` ?

File: pyproject.toml [hunk 7]
```diff
@@ -1,17 +1,132 @@
...
+    "ipykernel",
+    "packaging",
+    "tornado",
+    "jupyter-core",
+    "jupyterlab-server",
+    "notebook_shim",
+    "jupyterlab-widgets",
+    "nbformat",
+    "jupyter-core",
+    "numpy<2.0.2",
```
Comment (liadomide) [inline pyproject.toml]: I believe it should be `numpy<2`

File: xai_components/xai_connectivity/siibra_connectivity.py [hunk 8]
```diff
@@ -5,30 +5,28 @@
 # (c) 2022-2023, TVB Widgets Team
```
Comment (romina1601) [inline xai_components/xai_connectivity/siibra_connectivity.py]: Should we also update this text with 2024?

Comment (liadomide) [reply xai_components/xai_connectivity/siibra_connectivity.py]: yep. that would be an update on many files, but we should do it

**Review by romina1601 — APPROVED** [review]

**Pull request merged**